File: scrape_wc.py
import urllib.request
from bs4 import BeautifulSoup
import json

url = "https://en.wikipedia.org/wiki/2026_FIFA_World_Cup"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
html = urllib.request.urlopen(req).read()
soup = BeautifulSoup(html, 'html.parser')

# Find match tables
matches = []
for div in soup.find_all('div', class_='footballbox'):
    date_el = div.find('div', class_='fdate')
    time_el = div.find('div', class_='ftime')
    team1_el = div.find('th', class_='fhome')
    team2_el = div.find('th', class_='faway')
    stadium_el = div.find('div', itemprop='location')
    
    date = date_el.text.strip() if date_el else 'TBD'
    time = time_el.text.strip() if time_el else 'TBD'
    team1 = team1_el.text.strip() if team1_el else 'TBD'
    team2 = team2_el.text.strip() if team2_el else 'TBD'
    stadium = stadium_el.text.strip() if stadium_el else 'TBD'
    
    matches.append({
        'date': date,
        'time': time,
        'team1': team1,
        'team2': team2,
        'stadium': stadium
    })

print(f"Found {len(matches)} matches")
if len(matches) > 0:
    print(matches[:5])

File: scrape_wc.mjs
import * as cheerio from 'cheerio';
import fs from 'fs';

async function run() {
  const res = await fetch("https://en.wikipedia.org/wiki/2026_FIFA_World_Cup");
  const html = await res.text();
  const $ = cheerio.load(html);
  
  const matches = [];
  $('div.footballbox').each((i, el) => {
    const date = $(el).find('div.fdate').text().trim();
    const time = $(el).find('div.ftime').text().trim();
    const home = $(el).find('th.fhome').text().trim();
    const away = $(el).find('th.faway').text().trim();
    const venue = $(el).find('div[itemprop="location"]').text().trim();
    matches.push({ id: i+1, date, time, home, away, venue });
  });
  console.log(`Found ${matches.length} matches`);
  if (matches.length > 0) {
    console.log(matches.slice(0, 5));
    fs.writeFileSync('matches_wiki.json', JSON.stringify(matches, null, 2));
  }
}
run();
